src/hdf5_init.c

```diff
@@ -0,0 +1,177 @@
+#include "hdf5_init.h"
+#include "blosc_filter.h"
+
+#include <blosc.h>
+#include <hdf5.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+/*
+// functions for debugging
+void check_space_and_storage(hid_t dset) {
+        H5D_space_status_t space_status;
+        hsize_t storage_size;
+
+        status = H5Dget_space_status(dset, &space_status);
+        storage_size = H5Dget_storage_size(dset);
+        printf("Space for dataset has%sbeen allocated. \n", space_status ==
+H5D_SPACE_STATUS_ALLOCATED ? " " : " NOT "); printf("Storage size for dataset is
+: %ld bytes.\n", (long)storage_size);
+}
+*/
+
+// REAL functions
+void initialize_file_and_plist(char *filename,
+                               hid_t *file_id,
+                               hid_t *fapl_id,
+                               hid_t *fcpl_id)
+{
+  unsigned mode = H5F_ACC_TRUNC;
+  char *version, *date;
+
```

Comment:
Always perform a NULL check on all the pointers before using/dereferencing them.